utility/types/tx_fifo_mempool_test.go

```diff
@@ -0,0 +1,298 @@
+			},
+			wantItems: [][]byte{messageSendFactory(9), messageSendFactory(10)},
+		},
+		{
+			name: "RemoveTx should remove a transaction from the mempool",
+			args: args{
+				maxTxBytes: 1000,
+				maxTxs:     100,
+				initialTxs: &[][]byte{
+					messageSendFactory(9),
+					messageSendFactory(10),
+				},
+				actions: &[]func(*txFIFOMempool){
+					func(txFIFOMempool *txFIFOMempool) {
+						txFIFOMempool.RemoveTx(messageSendFactory(9))
+					},
+				},
+			},
+			wantItems: [][]byte{
+				messageSendFactory(10),
+			},
+		},
+		{
+			name: "Contains should return true for a transaction that is in the mempool and false viceversa",
+			args: args{
+				maxTxBytes: 1000,
+				maxTxs:     100,
+				initialTxs: &[][]byte{
+					messageSendFactory(9),
+					messageSendFactory(10),
+				},
+				actions: &[]func(*txFIFOMempool){
+					func(txFIFOMempool *txFIFOMempool) {
+						txHashOK := crypto.GetHashStringFromBytes(messageSendFactory(9))
+						require.True(t, txFIFOMempool.Contains(txHashOK), "mismatching Contains")
+						txHashKO := crypto.GetHashStringFromBytes(messageSendFactory(19))
+						require.False(t, txFIFOMempool.Contains(txHashKO), "mismatching Contains")
+					},
+				},
+			},
+			wantItems: [][]byte{
+				messageSendFactory(9),
+				messageSendFactory(10),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			txFIFOMempool := NewTxFIFOMempool(tt.args.maxTxBytes, tt.args.maxTxs)
+
+			if tt.args.initialTxs != nil {
+				for _, item := range *tt.args.initialTxs {
+					txFIFOMempool.AddTx(item)
+				}
+			}
+
+			if tt.args.actions != nil {
+				for _, action := range *tt.args.actions {
+					action(txFIFOMempool)
+				}
+			}
+
+			require.Equal(t, len(tt.wantItems), int(txFIFOMempool.TxCount()), "mismatching TxCount (capacity filled with elements)")
+
+			for _, wantItem := range tt.wantItems {
+				wantHash := crypto.GetHashStringFromBytes(wantItem)
+				require.True(t, txFIFOMempool.Contains(wantHash), "missing element")
+				gotItem, err := txFIFOMempool.PopTx()
+				require.NoError(t, err, "unexpected error while popping element")
+				require.Equal(t, wantItem, gotItem, "mismatching element")
+			}
+
+			if txFIFOMempool.TxCount() == 0 {
+				require.True(t, txFIFOMempool.IsEmpty(), "IsEmpty should return true when TxCount() is 0")
+			}
+		})
+	}
+}
+
+// messageSendFactory returns a MessageSend message marshalled in []byte format
+//
+// it is used in tests to make sure that the mempool is correctly limiting the size of the transactions it stores
+//
+// IMPORTANT: the size of the transaction is not the size of the whole message on the wire, there's obviously some overhead in the protobuf
+func messageSendFactory(size int) []byte {
+	amountBz := make([]byte, size)
+	for i := range amountBz {
+		amountBz[i] = 1
+	}
+
+	message := &MessageSend{
+		FromAddress: []byte{},
+		ToAddress:   []byte{},
+		Amount:      string(amountBz),
+	}
+	messageBz, _ := codec.GetCodec().Marshal(message)
```

Comment:
I think passing in `t` and running a `require.Error` here is fair game

Reply:
I think it's a bit overkill and would add a "distracting" element to the factory signature. Ultimately a panic would be telling and regardless if we can't marshal an empty message our codebase would look like this anyway:

![image](https://user-images.githubusercontent.com/29378614/216637285-a42d524e-1339-4357-8bff-8803da5edacf.png)

I hear you but I think it's clean enough and the corner we are cutting by not having a `require.NoError()` is negligeable.

If I didn't convince you with my hell picture I'll just update the code :)

Reply:
You convinced me with your hell picture. Will put it out by resolving this conversation 🚒